.github/workflows/build.yml

```diff
@@ -145,3 +145,17 @@ jobs:
         with:
           components: clippy
       - run: cargo clippy --all-features -- --deny warnings
+
+  features:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+      - name: Install cargo-hack
+        uses: taiki-e/install-action@cargo-hack
+      - name: Check each features
+        run: cargo hack --no-dev-deps check --each-feature
```

Comment:
WDYT about using `--feature-powerset ` instead of `--each-feature`? Would it inflate CI time too substantially?

Reply:
Updates to use `--feature-powerset` to test the CI time.

Reply:
In this test, the CI time increased from `32s` (https://github.com/tottoto/hyper-rustls/actions/runs/7050201648/job/19190355371) to `44s` (https://github.com/rustls/hyper-rustls/actions/runs/7050273761/job/19190655897). I think it is within an acceptable range as the impact on the overall CI is small.

Reply:
Agreed, that seems totally reasonable to me. Would you mind squashing the commit to use the powerset into the commit adding the job?

Reply:
Squashed the two commits.

Reply:
Thanks!